Tests/LibWeb/test-web/Fuzzy.cpp

```diff
@@ -0,0 +1,104 @@
+/*
+ * Copyright (c) 2025, Jelle Raaijmakers <jelle@ladybird.org>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include "Fuzzy.h"
+
+#include <LibGfx/Bitmap.h>
+
+namespace TestWeb {
+
+// https://web-platform-tests.org/writing-tests/reftests.html#fuzzy-matching
+bool fuzzy_screenshot_match(Gfx::Bitmap const& bitmap_a, Gfx::Bitmap const& bitmap_b, Vector<FuzzyMatch> const& fuzzy_matches)
+{
+    // No fuzzy matching: exact match.
+    auto diff = bitmap_a.diff(bitmap_b);
+    if (fuzzy_matches.is_empty())
+        return diff.identical;
+
+    // Apply each fuzzy match to the diff result.
+    for (auto const& fuzzy_match : fuzzy_matches) {
+        if (fuzzy_match.reference.has_value()) {
+            dbgln("FIXME: Fuzzy match restricted to '{}', but this is not implemented.", fuzzy_match.reference.value());
+            continue;
+        }
+
+        if (fuzzy_match.color_value_error.has_value()) {
+            auto color_error_range = fuzzy_match.color_value_error.value();
+            if (diff.maximum_error < color_error_range.minimum_value || diff.maximum_error > color_error_range.maximum_value)
+                return false;
+        } else if (diff.maximum_error > 0) {
+            return false;
+        }
+
+        if (fuzzy_match.pixel_count_error.has_value()) {
+            auto pixel_count_range = fuzzy_match.pixel_count_error.value();
+            if (diff.pixel_error_count < pixel_count_range.minimum_value || diff.pixel_error_count > pixel_count_range.maximum_value)
+                return false;
+        } else if (diff.pixel_error_count > 0) {
```

Comment:
It looks like if `maxDifference` is given but `totalPixels` is not the test will always fail here? I'm not sure what the intended behavior here is because there doesn't seem to be any examples in the WPT repo where only one value is set.

I'm not sure this is wrong, but I'm not sure how it's supposed to work.